azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -969,6 +976,43 @@ private void insertJobMetadata() {
         StringUtils.join2(this.node.getInNodes(), ","));
   }
 
+  /**
+   * Build Azkaban frontend base url. Use external configs
```

Comment:
Could you please elaborate more on why AZKABAN_WEBSERVER_URL does not work by default here?

Reply:
I think there is some redundancy of properties in the code to refer to the web server host. For example there is: `azkaban.server.hostname`, `jetty.hostname` and `azkaban.webserver.url`. I'm not sure what's the difference between them honestly and maybe we can get rid of `azkaban.webserver.url` at least since that one was only used in JobRunner class, the 2 other are more widely used throughout the code but it would be good to reduce usage to one only.

For user facing server url (which might be different from hostname and above properties if the web server sits behind a proxy for example) the code suggests to use `azkaban.webserver.external_hostname`, `azkaban.webserver.external_ssl_port` and `azkaban.webserver.external_port`, that is why we're using them here.

Reply:
From what I know:
1. jetty.hostname is the de-facto web server name but this config only exists in web server but not exec/pod
2. azkaban.server.hostname: it is either webserver/execserver hostname or pod name (fc-dep-<clustername>-<execid>) which is used to uniquely identity the physical machine.

Yep. Sorry I didn't make it clear. Could you please add 
`For user facing server url (which might be different from hostname and above properties if the web server sits behind a proxy for example) the code suggests to use azkaban.webserver.external_hostname, azkaban.webserver.external_ssl_port and azkaban.webserver.external_port, that is why we're using them here.`
into the code comment?

Reply:
`jetty.hostname` should be valid for both Web and Executor servers as there is a jetty web server running in the Executor.
IMO `azkaban.server.hostname` should not be the pod name as that is not a hostname but a logical application related name right? Based on code `azkaban.server.hostname` might be equivalent to Web server prop `jetty.hostname` in Executor server.
Both properties has `hostname` in their name which makes it difficult to distinguish their meaning and usage.

Reply:
Yeah. I understand the azkaban.server.hostname part. But that's how we did there LOL.
"Both properties has hostname in their name which makes it difficult to distinguish their meaning and usage.": can't agree more. Those props are pretty confusing!!!